validation/src/verifier/key_verifier.rs

```diff
@@ -0,0 +1,76 @@
+use anyhow::{anyhow, bail, Context, Result};
+use base64::engine::{self, Engine as _};
+use ceramic_core::Jwk;
+use ceramic_event::unvalidated::signed::Envelope;
+use tracing::{debug, warn};
+
+#[async_trait::async_trait]
+pub trait Verifier {
+    /// Verify the signature. Optoinally requiring it matches a delegated KID
+    async fn verify_signature(&self, delegated_kid: Option<&str>) -> Result<()>;
+}
+
+#[async_trait::async_trait]
+impl Verifier for Envelope {
+    async fn verify_signature(&self, delegated_kid: Option<&str>) -> Result<()> {
```

Comment:
Should return a valid event struct in the future. just putting this here so that we don't forget about it.

Reply:
yeah, we could add something in the future. I was holding off on a "validated event" because that requires more than just a signature (and this is a "can you trust what was inside this"). I was hoping to add that concept to the service in my next PR and make any changes here as needed.